src/controllers/url-controller.ts

```diff
@@ -0,0 +1,152 @@
+
+    const custom = customName && convertToHyphenated(customName);
+
+    const originalUrl = url;
+    const shortUrl = custom ? `https://shortit/${custom}` : `https://shortit/${urlString}`;
+    const customNameUrl = custom ? `${custom}` : '';
+
+    const shortenUrl = await prisma.url.create({
+      data: {
+        originalUrl,
+        shortUrl,
+        customName: customNameUrl,
+      },
+    });
+    return res.status(201).json({ message: successMsg, shortenUrl });
+  } catch (error) {
+    logger.info(error);
+    return res.status(400).send({ message: 'Error adding url' });
+  }
+};
+
+//  GET ALL THE URLS IN THE DATABASE
+export const getUrls = async (req: Request, res: Response) => {
+  try {
+    const url = await prisma.url.findMany({
+      orderBy: {
+        createdAt: 'desc',
+      },
+    });
+
+    if (!url) return res.status(404).send({ message: 'No url found!' });
+
+    return res.status(200).send({
+      message: successMsg,
+      result: url.length,
+      url,
+    });
+  } catch (error) {
+    logger.info(error);
+    return res.status(400).send({ message: 'Error getting url' });
+  }
+};
+
+//  GET A SINGLE URL FROM THE DATABASE
+export const getUrl = async (req: Request, res: Response) => {
+  const { id } = req.params;
+
+  try {
+    const url = await prisma.url.findUnique({
+      where: {
+        id,
+      },
+    });
+
+    if (!url) return res.status(404).send({ message: notFoundMsg });
+
+    return res.status(200).send({ message: successMsg, url });
+  } catch (error) {
+    logger.info(error);
+    return res.status(400).send({ message: 'Error occurred while fetching url' });
+  }
+};
+
+// UPDATING A SINGLE URL FROM THE DATABASE
+export const updateUrl = async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const { url, customName } = req.body;
+
+  if (!id) return res.status(404).send({ message: notFoundMsg });
+  if (!url && !customName) return res.status(400).send({ message: 'At least one field should be updated' });
+
+  const custom = customName && convertToHyphenated(customName);
+  const shortUrl = `https://shortit/${custom}`;
+
+  try {
+    const existingUrl = await prisma.url.findUnique({
+      where: {
+        id,
+      },
+    });
+
+    if (!existingUrl || existingUrl?.originalUrl !== url) {
+      return res.status(404).send({ message: notFoundMsg });
+    }
+
+    const updatedUrl = await prisma.url.update({
+      where: {
+        id,
+      },
+      data: {
+        customName: custom,
+        shortUrl,
+      },
+    });
+
+    return res.status(200).send({ message: 'Url updated successfully', updatedUrl });
+  } catch (error) {
+    logger.info(error);
+    return res.status(400).send({ message: 'Error occurred while updating url' });
+  }
+};
+
+// DELETING A SINGLE URL FROM THE DATABASE
+export const deleteUrl = async (req: Request, res: Response) => {
+  const { id } = req.params;
+  if (!id) return res.status(404).send({ message: notFoundMsg });
```

Comment:
I don't think we need to check for id. If there's no id, this route would not be found.

Reply:
okay boss